react-common/components/controls/MenuBar.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import { classList, ContainerProps } from "../util";
+
+export interface MenuBarProps extends ContainerProps {
+}
+
+export const MenuBar = (props: MenuBarProps) => {
+    const {
+        id,
+        className,
+        role,
+        ariaHidden,
+        ariaLabel,
+        children
+    } = props;
+
+    let focusableElements: HTMLElement[];
+    let menubar: HTMLDivElement;
+
+    const handleRef = (ref: HTMLDivElement) => {
+        if (!ref || menubar) return;
+
+        menubar = ref;
+
+        const focusable = ref.querySelectorAll(`[tabindex]:not([tabindex="-1"]),[data-isfocusable]`);
+        focusableElements = [];
+
+        for (const element of focusable.values()) {
+            focusableElements.push(element as HTMLElement);
+
+            // Remove them from the tab order, menu items are navigable using the arrow keys
+            element.setAttribute("tabindex", "-1");
+            element.setAttribute("data-isfocusable", "true");
+        }
+    }
+
+    const onKeyDown = (e: React.KeyboardEvent<HTMLElement>) => {
```

Comment:
also for later: we could consider moving this key nav stuff into a common util or react hook or something--not sure if there will be other components that use the arrow key navigation?